sfft_and_animate.py

```diff
@@ -1,300 +1,308 @@
+    sci_psf_path = get_imsim_psf(ra,dec,band,sci_pointing,sci_sca)
+    if verbose:
+        print('\n')
+        print('Path to science PSF:')
+        print(sci_psf_path)
+        print('\n')
+
+    t_imsim_psf = get_imsim_psf(ra,dec,band,t_pointing,t_sca)
+    t_psf_path = rotate_psf(ra,dec,t_imsim_psf,sci_skysub_path,force=True)
+    if verbose:
+        print('\n')
+        print('Path to template PSF:')
+        print(t_psf_path)
+        print('\n')
+
+    t_filepath = _build_filepath(None,band,t_pointing,t_sca,'image')
+    t_skysub = sky_subtract(path=t_filepath)
+    t_align = imalign(template_path=sci_skysub_path,sci_path=t_skysub) # NOTE: This is correct, not flipped.
+
+    overlap = check_overlap(ra,dec,t_align,sci_skysub_path,verbose=verbose)
+    if overlap:
+        sci_conv, temp_conv = crossconvolve(sci_skysub_path,sci_psf_path,t_align,t_imsim_psf,force=True)
+        if verbose:
+            print('\n')
+            print('Path to cross-convolved science image:')
+            print(sci_conv)
+            print('Path to cross-convolved template image:')
+            print(temp_conv)
+            print('\n')
+
+        diff_savename = f'{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits' # 'diff_' gets appended to the beginning of this
+        diff, soln = difference(sci_conv,temp_conv,sci_psf_path,t_psf_path,savename=diff_savename,backend='Numpy',force=True)
+        if verbose:
+            print('\n')
+            print('Path to differenced image:')
+            print(diff)
+            print('\n')
+        
+        dcker_savename = f'dcker_{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits'
+        dcker_path = decorr_kernel(sci_skysub_path,t_skysub,sci_psf_path,t_psf_path,diff,soln,savename=dcker_savename)
+        if verbose:
+            print('\n')
+            print('Path to decorrelation kernel:')
+            print(dcker_path)
+            print('\n')
+        
+        decorr_imgpath = decorr_img(diff,dcker_path)
+        if verbose:
+            print('\n')
+            print('Path to final decorrelated differenced image:')
+            print(decorr_imgpath)
+            print('\n')
+
+        zpt_savename = f'zptimg_{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits'
+        zpt_imgpath = decorr_img(sci_conv,dcker_path,savename=zpt_savename)
+        if verbose:
+            print('\n')
+            print('Path to zeropoint image:')
+            print(zpt_imgpath)
+            print('\n')
+
+        decorr_psf_savename = f'psf_{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits'
+        decorr_psfpath = decorr_img(sci_psf_path,dcker_path,savename=decorr_psf_savename)
+        if verbose:
+            print('\n')
+            print('Path to decorrelated PSF (use for photometry):')
+            print(decorr_psfpath)
+            print('\n')
+
+        return sci_skysub_path, decorr_imgpath
+
+    elif not overlap:
```

Comment:
Consider doing this check at the beginning, and just having the first bit say
```
if not overlap:
   [...]
    return None, None
```

and then you can include the bulk of the code of the function without having to wrap the main code within an "if" or "else".